yaksh/models.py

```diff
@@ -134,6 +140,346 @@ def create_trial_course(self, user):
     def get_hidden_courses(self, code):
         return self.filter(code=code, hidden=True)
 
+
+#############################################################################
+class Lesson(models.Model):
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    html_data = models.TextField(null=True, blank=True)
+    creator = models.ForeignKey(User)
+
+    def get_files(self):
+        return LessonFile.objects.filter(lesson=self)
+
+    def __str__(self):
+        return "{0}".format(self.name)
+
+
+#############################################################################
+class LessonFile(models.Model):
+    lesson = models.ForeignKey(Lesson, related_name="lesson")
+    file = models.FileField(upload_to=get_file_dir)
+
+    def remove(self):
+        if os.path.exists(self.file.path):
+            os.remove(self.file.path)
+            if os.listdir(os.path.dirname(self.file.path)) == []:
```

Comment:
Just curious, why is this needed?

Reply:
This is to check if the directory is empty so as to delete the directory if the directory is empty.